main.go

```diff
@@ -45,6 +46,29 @@ func main() {
 	arn := ""
 	var err error
 
+	if *requirelatest {
+		// check if the latest task definition is running. if it is not, do not proceed with the deployment.
+		taskDef, err := c.GetTaskDefinition(task)
+		if err != nil {
+			logger.Printf("[error] get task definition: %s\n", err)
+			os.Exit(1)
+		}
+
+		deployments, err := c.GetDeployments(cluster, service)
+		if len(deployments) != 1 {
+			logger.Printf("[error] not exactly one deployment found: %d\n", len(deployments))
+			os.Exit(1)
+		}
```

Comment:
this can actually happen if there are two deploys going on in close proximity... possibly not worth worrying about, but it's something else to keep in mind

Reply:
yeah good call i'll call that out. it's possible to figure out which deployment is target for primary but if the circuit breaker trips that's no longer true.